wicket/src/ui/panes/update.rs

```diff
@@ -829,8 +841,150 @@ impl UpdatePane {
+                };
+
+                if help_text.is_some() {
+                    // message_rect.height needs to be subtracted by 3 to
+                    // account for the box being moved downards (message_rect.y
+                    // += 3).
+                    message_rect.height = message_rect.height.saturating_sub(3);
+                } else {
+                    // As above, message_rect.height needs to be subtracted by 3
+                    // to account for the box being moved downwards. *However*,
+                    // the height needs to be incremented by 3 since there's no
+                    // help bar. This means that there is no change in height.
+                }
+
+                let body = match response {
+                    Ok(()) => {
+                        // Don't display any text here; status_text should be
+                        // enough for the user.
+                        Text::from(Vec::new())
+                    }
+                    Err(message) => {
+                        // Print out the message here.
+                        //
+                        // TODO: show a list of causes -- HttpError (which this
+                        // ultimately comes from) doesn't seem to have a good
+                        // way to return structured data like that.
+                        let mut spans = Vec::new();
+
+                        // If the message has multiple lines of text, split them
+                        // into separate spans. This makes text wrapping offsets
+                        // work correctly.
+                        let mut next_line =
+                            vec![Span::styled("Message: ", style::selected())];
+                        for line in message.lines() {
+                            next_line.push(Span::styled(
+                                line.to_owned(),
+                                style::plain_text(),
+                            ));
+                            spans.push(Spans::from(next_line));
+                            next_line = Vec::new();
+                        }
+                        spans.push(Spans::from(next_line));
+
+                        Text::from(spans)
+                    }
+                };
+
+                let paragraph = Paragraph::new(status_text)
+                    .block(block.clone().title("UPDATE STATUS"))
+                    .alignment(Alignment::Center);
+                frame.render_widget(paragraph, status_text_rect);
+
+                // Wrap the text to the screen width.
+                let options = crate::ui::wrap::Options {
+                    // Subtract 2 for borders.
+                    width: message_rect.width.saturating_sub(2) as usize,
+                    initial_indent: Span::raw(""),
+                    subsequent_indent: Span::raw(""),
+                    break_words: true,
+                };
+                let wrapped_body = wrap_text(&body, options);
```

Comment:
Are we assuming for now that this text doesn't exceed the height of `message_rect`? If so, I guess we'll add scrolling in the future *if* we need it.

Reply:
Ah yeah, we're going to need scrolling here at some point maybe. I think error messages tend to be a bit short.